src/client-agent/notify.c

```diff
@@ -177,7 +167,7 @@ void run_notify()
         }
     }
     /* Create message */
-    if(*agent_ip != '\0' && strcmp(agent_ip, "Err")) {
+    if (*agent_ip != '\0') {
```

Comment:
The line 161 still needs to be fixed.
Even on error, the `strdup()` at the end of `get_agent_ip()` will always return a valid pointer, so the **else** block will never be executed.

Reply:
strdup could return NULL if ENOMEM errno, and get_agent_ip could return '\0' if another problem appears.

Reply:
Ok, thank you.